onnxruntime/core/framework/allocation_planner.h

```diff
@@ -71,9 +76,45 @@ class SequentialPlannerContext : public ISequentialPlannerContext {
   bool enable_memory_reuse_ = true;
 };
 
+class ParallelPlannerContext : public SequentialPlannerContext {
```

Comment:
This hierarchy is a bit confusing. Does it make sense to have a IParallelPlannerContext? or if things are common, IPlannerContext?